src/test.rs

```diff
@@ -363,4 +365,55 @@ impl TestRequest {
     {
         block_on(f)
     }
+
+    /// Helper function that returns a deserialized response body of a TestRequest
+    /// This function blocks the current thread until futures complete.
+    ///
+    /// ```rust
+    /// use actix_web::{App, test};
+    /// use serde::{Serialize, Deserialize};
+    /// 
+    /// #[derive(Serialize, Deserialize)]
+    /// pub struct Person { id: String, name: String }
+    /// 
+    /// #[test]
+    /// fn test_add_person() {
+    ///     let mut app = test::init_service(App::new().service(
+    ///                              web::resource("/people")
+    ///                                .route(web::post().to(|person: Json<Person>| {
+    ///                                       HttpResponse::Ok()
+    ///                                          .json(person.into_inner())})
+    ///                                 )));
+    /// 
+    ///     let payload = r#"{"id":"12345","name":"Nikolay Kim"}"#.as_bytes();
+    /// 
+    ///     let req = test::TestRequest::post()
+    ///                      .uri("/people")
+    ///                      .header(header::CONTENT_TYPE, "application/json")
+    ///                      .set_payload(payload)
+    ///                      .to_request();
+    /// 
+    ///     let result = test::read_response_json<_, _, _, Person>(&mut app, req);
+    /// }
+    /// ```
+    pub fn read_response_json<S, R, B, T>(app: &mut S, req: R) -> T
+    where
+        B: MessageBody,
+        S: Service<Request = R, Response = ServiceResponse<B>, Error = Error>,
```

Comment:
you can also replace `R` with `ServiceRequest`

Reply:
done